src/analyzer/protocol/dns/DNS.cc

```diff
@@ -1644,17 +1644,19 @@ bool DNS_Interpreter::ParseRR_AAAA(detail::DNS_MsgInfo* msg,
 bool DNS_Interpreter::ParseRR_WKS(detail::DNS_MsgInfo* msg,
                                   const u_char*& data, int& len, int rdlength)
 	{
-	data += rdlength;
-	len -= rdlength;
-
-	return true;
-	}
+	if ( ! dns_WKS_reply || msg->skip_event )
+		{
+		data += rdlength;
+		len -= rdlength;
+		return true;
+		}
 
-bool DNS_Interpreter::ParseRR_HINFO(detail::DNS_MsgInfo* msg,
-                                    const u_char*& data, int& len, int rdlength)
-	{
-	data += rdlength;
-	len -= rdlength;
+	// TODO: Pass the ports as parameters to the event
+	analyzer->EnqueueConnEvent(dns_WKS_reply,
+		analyzer->ConnVal(),
+		msg->BuildHdrVal(),
+		msg->BuildAnswerVal()
+	);
 
 	return true;
```

Comment:
The other change checks the length of rdlength as the return value. Should this one do the same?

Reply:
Honestly, I'm not sure. It seems like the current functions have a mix of `return true` and `return rdlength == 0`. Because I'm not actually parsing those fields in WKS replies right now, I just hardcoded a perma-true status. Maybe leave it as-is and add a comment to update that to actually reflect the status if the fields ever get parsed?

Reply:
Yah that sounds fine to me.